### DISCUSSION TITLE
Can skywalking count real-time jvm memory usage?

### DISCUSSION THREAD

**Discussion by mark8s** [apache/skywalking#11584] [General]
I know that skwalking-ui can display some jvm statistics, but I did not find the jvm memory usage. I would like to know if there is a way to obtain this value in skywalking?

I hope you can help me answer it

**Comment by kezhenxu94** [comment]
We have collected and display the JVM memory there, can you be more clear what do you want?


<img width="1898" alt="image" src="https://github.com/apache/skywalking/assets/15965696/ad48d6fa-c75a-49f3-b58f-d95a7d863b4a">

**Reply by mark8s** [reply]
I want to get the jvm memory usage %, these charts don't quite fit

**Reply by kezhenxu94** [reply]
Learn our query expression and it's quite easy to add a new line in the chart to plot the percentage with expression `instance_jvm_memory_heap / instance_jvm_memory_heap_max`

**Reply by mark8s** [reply]
ok , thanks for your reply, I'll try it first

**Reply by mark8s** [reply]
@kezhenxu94 Is there any ready-made indicator for calculating jvm memory usage? I want to check it out at once through graphsql, instead of getting the instance_jvm_memory_heap_max and instance_jvm_memory_heap and then calculating the data through the code program. Is there any way?

**Reply by kezhenxu94** [reply]
> @kezhenxu94 Is there any ready-made indicator for calculating jvm memory usage? I want to check it out at once through graphsql, instead of getting the instance_jvm_memory_heap_max and instance_jvm_memory_heap and then calculating the data through the code program. Is there any way?

Please, what I mentioned above is through graphql query, you don't need to get that through the code program unless you want. Read the doc please.

**Reply by wu-sheng** [reply]
Make sure you are using latest release. metric expression(MQE in API doc) is newly added.(9.6+)

**Reply by mark8s** [reply]
@kezhenxu94 @wu-sheng Thanks for both of your replies. I'll take a closer look

**Reply by mark8s** [reply]
@kezhenxu94 Hello, is that so?
![image](https://github.com/apache/skywalking/assets/47961301/40a0edc1-a10b-4b46-ab3e-8884ab3cbb3c)

**Reply by kezhenxu94** [reply]
yes

**Reply by mark8s** [reply]
@kezhenxu94 tks~
